src/components/tree-detail/last-waterings/watering-card.tsx

```diff
@@ -25,12 +25,15 @@ export const WateringCard: React.FC<WateringCardProps> = ({ wateringData }) => {
 	const formatDate = useI18nStore().formatDate;
 	const { username } = useProfileStore();
 	const { deleteWatering } = useWaterTree();
+	const [isDeleteWateringLoading, setIsDeleteWateringLoading] = useState(false);
 
 	const [isConfirmDeleteVisible, setIsConfirmDeleteVisible] = useState(false);
 	const isWateringByUser = wateringData.username === username;
 
 	const onClickDelete = async () => {
+		setIsDeleteWateringLoading(true);
 		await deleteWatering(wateringData.id);
+		setIsDeleteWateringLoading(false);
```

Comment:
I'd suggest to remove the `wateringLoading` state inside the `useWaterTree()` hook at the same time 👍